api/test/json-api.e2e-spec.ts

```diff
@@ -0,0 +1,44 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { INestApplication } from '@nestjs/common';
+import * as request from 'supertest';
+import { AppModule } from '../src/app.module';
+
+describe('JSON API Specs (e2e)', () => {
+  let app: INestApplication;
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    await app.init();
+  });
+
+  afterAll(async () => {
+    await Promise.all([app.close()]);
+  });
+  it('should return a response shaped as JSON:API Error spec', async () => {
+    const jsonApiErrorResponse = {
+      status: null,
+      title: null,
+      meta: {
+        timestamp: null,
+        path: null,
+        type: null,
+      },
+    };
+    const response = await request(app.getHttpServer())
+      .post('/auth/sign-in')
+      .send({
+        username: 'fakeuser@example.com',
+        password: 'fakePassword',
+      });
+    response.body.errors.forEach((err: any) => {
+      expect(Object.keys(err)).toEqual(Object.keys(jsonApiErrorResponse));
```

Comment:
All looks good to me in this PR, thanks! my only question would be about this bit specifically.

Here we check that the root props of the error response object(s) match the list of three props in the `jsonApiErrorResponse` const above - which is fine. Since the JSON:API spec also allows to include `id`, `links`, `code`, `detail` and `source`, do we want to "whitelist" them in this test? This way, if we added allowed props later on to our response, the test will still pass (and correctly so) without needing to change it.

I am ok with the current implementation, just checking what your opinion would be on this.

Reply:
Yes I see your point. And I'll be happy to add this whitelist, but I didn't at the first place and here is the reasoning behind. I think that testing needs to be accurate to what logic/functionality we are covering with it, so if our production code does not send said properties I we shouldnt test that something 'might exist or not', but in the other hand, we are dealing with a standar specification and not an open interpretation of what we want to send, and suddenly does not seem that bad.
I'll add this one, thanks!